zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -312,7 +312,7 @@ public boolean runParagraph(String noteId,
     p.setConfig(config);
 
     if (note.isPersonalizedMode()) {
-      p = note.getParagraph(paragraphId);
+      p = p.getUserParagraph(context.getAutheInfo().getUser());
```

Comment:
I looked at the logic of the getUserParagraph code.
By copying a Paragraph, You can avoid multiple people working on the same Paragraph at the same time.